app/api/storage/upload-by-key/route.ts

```diff
@@ -0,0 +1,46 @@
+import { NextResponse } from "next/server";
+import supabase from "@/lib/supabase/serverClient";
+import { SUPABASE_STORAGE_BUCKET } from "@/lib/consts";
+
+function isValidKey(key: string): boolean {
+  if (!key || key.length > 1024) return false;
+  if (key.startsWith("/")) return false;
+  if (key.includes("..")) return false;
+  if (key.includes("\\")) return false;
+  return true;
+}
```

Comment:
Single responsibility principle
- actual: multiple functions defined in the same file
- required: one function defined per file